### PR TITLE
Fix issue with invoke pattern being provided for top navview item

### LINKED ISSUE #2702: NavigationViewItem tab control type supports invoke pattern with PaneDisplayMode="Top"
**Describe the bug**

`NavigationViewItem`s in a `NavigationView` with the `PaneDisplayMode` set to `Top` supports the invoke pattern, which is a violation of [Section 508 502.3.10](https://www.access-board.gov/guidelines-and-standards/communications-and-it/about-the-ict-refresh/final-rule/single-file-version#502-interoperability-assistive-technology) (according to Accessibility Insights for Windows).

**Steps to reproduce the bug**

1. Create an app with a `NavigationView` control with `PaneDisplayMode` set to `Top`.
2. Add `NavigationViewItems` in `NavigationView.MenuItems` with any content value: (ex: "Menu Item 1", "Menu Item 2", "Menu Item 3".
3. Run the app with Accessibility Insights.

**Expected behavior**

Expected: Invoke pattern is not supported.

**Screenshots**

**Version Info**

NuGet package version: 

[Microsoft.UI.Xaml 2.4.2]

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| November 2019 Update (18363)  | Yes   |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | Yes   |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Additional context**

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

Added check to ensure we only provide InvokePattern when we are not in top NavigationView
## Motivation and Context

Fixes #2702 
## How Has This Been Tested?

Tested manually:

image

Update API test

### CODE REVIEW

**Comment by StephenLPeters** [comment]
@chingucoding these test failures look real, the top nav tests appear to be expecting to be able to invoke the item while in top nav.

**Comment by chingucoding** [comment]
Right, thanks for the ping! I'll take a look tomorrow.

**Comment by chingucoding** [comment]
Tests should run successfully now.

File: dev/NavigationView/NavigationViewItemAutomationPeer.cpp [hunk 1]
```diff
@@ -47,7 +47,8 @@ winrt::hstring NavigationViewItemAutomationPeer::GetNameCore()
 winrt::IInspectable NavigationViewItemAutomationPeer::GetPatternCore(winrt::PatternInterface const& pattern)
 {
     if (pattern == winrt::PatternInterface::SelectionItem ||
-        pattern == winrt::PatternInterface::Invoke ||
+        // Only provide Invoke pattern when we are a ListItem i.e. not on top navigation
```
Comment (ranjeshj) [inline dev/NavigationView/NavigationViewItemAutomationPeer.cpp:50]: > not on top navigation [](start = 66, length = 22)

From the issue comments, it sounds like we don't want to support invoke on top or left navigation.

Comment (chingucoding) [reply dev/NavigationView/NavigationViewItemAutomationPeer.cpp:50]: Hmm I see. So should NavigationView be "Tab" control in left and top mode then too?

Comment (chingucoding) [reply dev/NavigationView/NavigationViewItemAutomationPeer.cpp:50]: @YuliKl What do you think?

Comment (YuliKl) [reply dev/NavigationView/NavigationViewItemAutomationPeer.cpp]: From what I read, Invoke and SelectionItem patterns are supposed to be mutually exclusive. And we need nav view items to support the SelectionItem pattern, so I believe that means removing the Invoke pattern regardless of NavigationView's display mode.

Comment (chingucoding) [reply dev/NavigationView/NavigationViewItemAutomationPeer.cpp]: I see, thanks for the clarification. This PR is inline with that requirement in its current state.

**Comment by chingucoding** [comment]
There were test failures due to the fact that some tests expected the NavView items to be invokable. Switched to `Click` and updated the settings invoke test, to test whether clicking works correctly.

**Comment by chingucoding** [comment]
It seems that the CI bugged out here. Can we rerun the CI?

**Pull request merged**